aiida_quantumespresso/workflows/pdos.py

```diff
@@ -0,0 +1,564 @@
+# -*- coding: utf-8 -*-
+"""Workchain to run Quantum ESPRESSO calculations that generate DoS and PDoS for a structure.
+
+This requires four computations:
+
+- SCF (pw.x), to generate the initial wavefunction.
+- NSCF (pw.x), to generate eigenvalues, generally with a denser k-point mesh and tetrahedra occupations.
+- Total DoS (dos.x), to generate total densities of state.
+- Partial DoS (projwfc.x), to generate partial densities of state, by projecting wavefunctions onto atomic orbitals.
+
+Additional functionality:
+
+- Setting ``'align_to_fermi': True`` in the input ``parameters`` node,
+  will ensure that the energy range is centred around the Fermi energy.
+
+Storage memory management:
+
+The wavefunction file(s) created by the nscf calculation can get very large (>100Gb).
```

Comment:
Will do 👍

Reply:
Just FYI, in [this branch](https://github.com/greschd/aiida-quantumespresso/tree/feature/allow_symlink_pp) I've implemented symlinking for `pp.x`. It's depends on https://github.com/aiidateam/aiida-core/pull/4727 though, which is why I haven't opened a PR yet.

The reason why that `aiida-core` change is needed is because it could _add_ files into the `out` directory, so each file needs to be symlinked separately -- instead of symlinking `out` as a whole. I haven't checked, but `dos` and `projwfc` might be similar.